source/schwarz_base.cpp

```diff
@@ -391,6 +438,38 @@ void SchwarzBase<ValueType, IndexType>::run(
                 4, metadata.my_rank, expand_local_vec, metadata.iter_count);
         }
     }
+
+    //CHANGED
+    //Closing file
+    fps.close();
+    fpr.close();
+   
+    //adding 1 to include the 0-th iteration
+    metadata.iter_count = metadata.iter_count + 1;
+
+    //number of messages a PE would send out without event-based
+    int noevent_msg_count = metadata.iter_count * num_neighbors_out;
+
+    int total_events = 0;
+
+    //Printing msg count
+    for (int k = 0; k < num_neighbors_out; k++) {
+	std::cout << " Rank: " << metadata.my_rank << " to "
+	      << neighbors_out[k] << " : " << this->comm_struct.msg_count->get_data()[k];
+        total_events += this->comm_struct.msg_count->get_data()[k];
+    }
+    std::cout << std::endl;
+
+    //Total no of messages in all PEs
+    MPI_Allreduce(MPI_IN_PLACE, &total_events, 1, MPI_INT, MPI_SUM, MPI_COMM_WORLD);
```

Comment:
These Allreduces will synchronize the whole solver and you will essentially have a synchronous solver.

Reply:
This is outside the timed section now.